components/claim/components/email_to_ldap.tsx

```diff
@@ -0,0 +1,241 @@
+        if (!ldapId) {
+            setLdapError(localizeMessage('claim.email_to_ldap.ldapIdError', 'Please enter your AD/LDAP ID.'));
+            setPasswordError('');
+            setLdapPasswordError('');
+            setServerError('');
+            return;
+        }
+
+        const ldapPassword = ldapPasswordInput.current?.value;
+        if (!ldapPassword) {
+            setLdapPasswordError(localizeMessage('claim.email_to_ldap.ldapPasswordError', 'Please enter your AD/LDAP password.'));
+            setLdapError('');
+            setPasswordError('');
+            setServerError('');
+            return;
+        }
+
+        setPassword(password);
+        setLdapId(ldapId);
+        setLdapPassword(ldapPassword);
+
+        if (email) {
+            submit({loginId: email, password, ldapIdParam: ldapId, ldapPasswordParam: ldapPassword});
+        }
+    };
+
+    const submit = ({loginId, password, token = '', ldapIdParam = '', ldapPasswordParam = ''}: SubmitOptions) => {
+        emailToLdap(
+            loginId,
+            password,
+            token,
+            ldapIdParam || ldapId,
+            ldapPasswordParam || ldapPassword,
+            (data: AuthChangeResponse) => {
+                if (data.follow_link) {
+                    window.location.href = data.follow_link;
+                }
+            },
+            (err: {server_error_id: string; id: string; message: string}) => {
+                if (!showMfa && err.server_error_id === ClaimErrors.MFA_VALIDATE_TOKEN_AUTHENTICATE) {
+                    setShowMfa(true);
+                } else {
+                    switch (err.id) {
+                    case ClaimErrors.ENT_LDAP_LOGIN_USER_NOT_REGISTERED:
+                    case ClaimErrors.ENT_LDAP_LOGIN_USER_FILTERED:
+                    case ClaimErrors.ENT_LDAP_LOGIN_MATCHED_TOO_MANY_USERS:
+                        setLdapError(err.message);
+                        setShowMfa(false);
+                        break;
+                    case ClaimErrors.ENT_LDAP_LOGIN_INVALID_PASSWORD:
+                        setLdapPasswordError(err.message);
+                        setShowMfa(false);
+                        break;
+                    case ClaimErrors.API_USER_INVALID_PASSWORD:
+                        setPasswordError(err.message);
+                        setShowMfa(false);
+                        break;
+                    default:
+                        setServerError(err.message);
+                        setShowMfa(false);
+                    }
+                }
+            },
+        );
+    };
+    const serverErrorElement: JSX.Element | null = <ErrorLabel errorText={serverError}/>;
+    const passwordErrorElement: JSX.Element | null = <ErrorLabel errorText={passwordError}/>;
+    const ldapErrorElement: JSX.Element | null = <ErrorLabel errorText={ldapError}/>;
+    const ldapPasswordErrorElement: JSX.Element | null = <ErrorLabel errorText={ldapPasswordError}/>;
```

Comment:
great work ... you can now add it to the return value directly without adding new variables for each of them... it reduces complexity even more and is more readable

Reply:
this comment applies to the other files as well